packages/domain/lib/src/use_cases/validate_login_form.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:domain/domain.dart';
+import 'package:domain/src/const/validation_and_login_result.dart';
+import 'package:domain/src/use_cases/base/in_out_use_case.dart';
+
+class ValidateLoginFormUseCase
+    implements InOutUseCase<UserEmailPass, ValidationResult> {
+  @override
+  ValidationResult call(UserEmailPass form) => ValidationResult(
+        login: _validateLogin(form.login),
+        password: _validatePassword(form.password),
+      );
+
+  String? _validateLogin(String login) {
+    if (_validate(_emptyStringPattern, login))
+      return ValidationAndLogin.loginIsRequired;
+    return (_validate(_loginFormatPattern, login))
+        ? null
+        : ValidationAndLogin.invalidLoginFormat;
+  }
+
+  String? _validatePassword(String password) {
+    if (_validate(_emptyStringPattern, password))
+      return ValidationAndLogin.passwordIsRequired;
+    return _validate(_passwordPattern, password)
+        ? null
+        : ValidationAndLogin.invalidPassword;
+  }
+
+  bool _validate(String pattern, String value) {
```

Comment:
let's validate not only via regex, but with some string checks. for ex check for required value - `value.trim().isNotEmpty()`, min length - `value.length` and so on